Escape HipChat room name

tkanzakic opened the pull request:
<!-- Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] -->

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
Fixes #10178 

### Description
The channel is being escaped before building the HipChat API URL for sending message to a specific user, but this was not true for messages sent to rooms.

Commit:
Escape HipChat room name

The channel is being escaped before building the HipChat API URL
for sending message to a specific user, but this was not true for
messages sent to rooms

tkanzakic commented:
I signed it!

ohayon commented on fastlane/lib/fastlane/actions/hipchat.rb:
```diff
@@ -46,19 +46,19 @@ def self.run(options)
           end
         else
           ########## running on V2 ##########
+          escaped_channel = URI.unescape(channel) == channel ? URI.escape(channel) : channel
```
Do you think we could simply use `channel.shellescape`?

tkanzakic replied:
Hi @ohayon, many thanks for your review, it is a big pleasure for me to contribute to the project. 

About your comment, `channel.shellescape` encode the string to be a valid shell resource and in this case we need a string encoded to be a valid URL resource, in "code words":

```ruby
require 'shellwords'
require 'uri'

channel = "Room name"
print(channel.shellescape) # prints "Room\\ name"
print(URI.escape(channel)) # prints "Room%20name"
```

The complexity of the line is because encoding an already encoded string produce a new value:

```ruby
print(URI.escape(URI.escape(channel))) # prints "Room%2520name"
```

for that reason we first verify if the `channel` string is already encoded or not.

ohayon replied:
Got it! Thanks for the explanation, that makes good sense. Do you think you could just add a little comment to make it a bit more clear and then we can get this merged in? 🚀  🐙

tkanzakic replied:
Done!

ohayon reviewed (commented):
Hey @tkanzakic! This is a great change! Thanks for the contribution. Just one comment above to see if we can make that line a bit more simple?

Thanks again! 🚀

Commit:
Add explanation comment (#10178)
Add comment to clarify why we are escaping the channel name and how the
URI.escape method works

ohayon commented:
Thanks @tkanzakic !! Great first contribution! Welcome to the _fastlane_ community! 🚀

tkanzakic commented:
@ohayon awesome, thanks!
